core/src/main/java/hudson/slaves/JNLPLauncher.java

```diff
@@ -195,6 +198,59 @@ public void launch(SlaveComputer computer, TaskListener listener) {
     @Restricted(NoExternalUse.class)
     public static /*almost final*/ Descriptor<ComputerLauncher> DESCRIPTOR;
 
+    @NonNull
+    @Restricted(NoExternalUse.class)
+    public String getRemotingOptionsUnix(@NonNull Computer computer) {
+        return getRemotingOptions(escapeUnix(computer.getName()));
+    }
+
+    @NonNull
+    @Restricted(NoExternalUse.class)
+    public String getRemotingOptionsWindows(@NonNull Computer computer) {
+        return getRemotingOptions(escapeWindows(computer.getName()));
+    }
+
+    private String getRemotingOptions(String computerName) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("-name ");
+        sb.append(computerName);
+        sb.append(' ');
+        if (isWebSocket()) {
+            sb.append("-webSocket ");
+        }
+        if (tunnel != null) {
```

Comment:
Same here. Also `workDirSettings`. IOW there is no longer a reason to have _any_ configurable parameters on `JNLPLauncher`.